packages/fx-core/src/plugins/resource/frontend/dotnet/plugin.ts

```diff
@@ -93,6 +96,30 @@ export class DotnetPluginImpl implements PluginImpl {
   }
 
   public async scaffold(ctx: PluginContext): Promise<TeamsFxResult> {
+    Logger.info(Messages.StartScaffold(PluginInfo.displayName));
+
+    const selectedCapabilities = (ctx.projectSettings?.solutionSettings as AzureSolutionSettings)
+      .capabilities;
+    const includeTab = selectedCapabilities.includes(DotnetSupportCapability.tabCapability);
+    const includeBot = selectedCapabilities.includes(DotnetSupportCapability.botCapability);
+    const projectName = ctx.projectSettings!.appName;
+
+    const templateVariable: TemplateVariable = { BlazorAppServer: projectName };
+
+    if (includeTab) {
+      templateVariable.IS_TAB = "true";
+      const templateInfo = new TemplateInfo(ctx, Group.Tab, templateVariable);
+      await Scaffold.scaffoldFromZipPackage(ctx.root, templateInfo);
+    }
+    if (includeBot) {
+      templateVariable.IS_BOT = "true";
+      const templateInfo = new TemplateInfo(ctx, Group.Bot, templateVariable);
```

Comment:
Is that expected that if we choose tab & bot, then in L116 templateVariable has both IS_TAB and IS_BOT is true? These two flags are not isolated according to this code. So it looks like templateVariable cannot be shared?

Reply:
Sure, this templateVariable will add the parameters as we expected.